flang/test/Lower/dummy-procedure-character.f90

```diff
@@ -0,0 +1,172 @@
+! Test lowering of character function dummy procedure. The length must be
+! passed along the function address.
+! RUN: bbc -emit-fir %s -o - | FileCheck %s
+
+! -----------------------------------------------------------------------------
+!     Test passing a character function as dummy procedure
+! -----------------------------------------------------------------------------
+
+! CHECK-LABEL: func @_QPcst_len
+subroutine cst_len()
+  interface
+    character(7) function bar1()
+    end function
+  end interface
+! CHECK:  %[[VAL_0:.*]] = fir.address_of(@_QPbar1) : (!fir.ref<!fir.char<1,?>>, index) -> !fir.boxchar<1>
```

Comment:
The type should have a `7` in the length field, it seems.

Reply:
I agree. I added a commit to this to ensure constant length is propagated (since the issue was not introduced by this patch, there is a few small modification to existing test cases).